application/src/main/java/org/togetherjava/tjbot/features/help/HelpSystemHelper.java

```diff
@@ -355,4 +357,39 @@ public String getTagName() {
             return tagName;
         }
     }
+
+    /**
+     * Retrieves a {@link HelpThreadsRecord} object from the database based on the specified channel
+     * ID.
+     * <p/>
+     * This method queries for a record in the HELP_THREADS table where the CHANNEL_ID column
+     * matches the provided channelID param. It returns an Optional containing the matching record.
+     *
+     * @param channelID is channel unique identifier
+     *
+     * @return An Optional containing the HelpThreadsRecord for the specified channel ID, or an
+     *         empty Optional if no matching record is found.
+     *
+     * @throws DataAccessException if something is wrong with database
+     *
+     */
+    Optional<HelpThreadsRecord> getThreadRecordByChannelId(final long channelID) {
+
+        logger.debug("Looking for thread-record using channel ID: {}", channelID);
+
+        try {
+
+            return database.read(ctx -> ctx.selectFrom(HELP_THREADS)
+                .where(HELP_THREADS.CHANNEL_ID.eq(channelID))
+                .limit(1)
+                .fetchOptional());
+
+        } catch (DataAccessException ex) {
+
+            logger
+                .error("Something went wrong while trying to fetch data from thread-record table");
+
```

Comment:
Do we need log level ERROR here? maybe warn or lower since we're already handling it.

Reply:
Hmm, well if it's **handled**, maybe the whole `try..catch` is unnecessary 🤔

Reply:
@ankitsmt211 i've changed it into a warning.

Reply:
if we drop the try-catch, the exception `NoSuchElement` exception should be thrown when method is called right?

Reply:
Sure. That exception means that no element has been found in database. Im going to remove the `try..catch` block.